test/eth/Cdp.spec.js

```diff
@@ -11,60 +11,78 @@ import {
   MKR
 } from '../../src/eth/Currency';
 import { promiseWait } from '../../src/utils';
-import { dappHub } from '../../contracts/abis';
-import testnetAddresses from '../../contracts/addresses/testnet.json';
 import { mineBlocks } from '../helpers/transactionConfirmation';
 
-let cdpService, cdp, currentAccount, dai, dsProxyAddress, smartContractService;
+let cdpService,
+  cdp,
+  currentAccount,
+  dai,
+  dsProxyAddress,
+  proxyAccount,
+  proxyKey,
+  transactionCount;
 
 // this function should be called again after reverting a snapshot; otherwise,
 // you may get errors about account and transaction nonces not matching.
-async function init() {
+async function init(proxy = false) {
   cdpService = buildTestEthereumCdpService();
   await cdpService.manager().authenticate();
-  currentAccount = cdpService
+  if (proxy) {
+    await setProxyAccount();
+  } else {
+    currentAccount = cdpService
+      .get('token')
+      .get('web3')
+      .currentAccount();
+  }
+}
+
+async function setProxyAccount(newAccount = false) {
+  const accountService = cdpService
     .get('token')
     .get('web3')
-    .currentAccount();
-  if (cdp) cdp._cdpService = cdpService;
+    .get('accounts');
+  const account = newAccount
+    ? {
+        address: newAccount.address,
+        key: newAccount.key
+      }
+    : {
+        address: proxyAccount,
+        key: proxyKey
+      };
+
+  await accountService.addAccount(account.address, {
+    type: 'privateKey',
+    key: account.key
+  });
+  accountService.useAccount(account.address);
+  currentAccount = account.address;
 }
 
-beforeAll(async () => {
-  await init();
-  dai = cdpService.get('token').getToken(DAI);
-  smartContractService = cdpService.get('smartContract');
-
-  // Clear owner of DSProxy created during testchain deployment
-  // (allowing us to create new DSProxy instances using the default address)
-  const owner = await getDsProxyOwner(testnetAddresses.DS_PROXY);
-  if (owner !== '0x0000000000000000000000000000000000000000') {
-    await clearDsProxyOwner(testnetAddresses.DS_PROXY);
-  }
-});
-
-afterAll(async () => {
-  await clearDsProxyOwner(dsProxyAddress);
-});
-
-function getDsProxy(address) {
-  return smartContractService.getContractByAddressAndAbi(
-    address,
-    dappHub.dsProxy,
-    { name: 'DS_PROXY' }
-  );
+async function transferMkr() {
```

Comment:
please have this take an argument rather than depending on a global, for explicitness. i'm asking for these changes because there are now a lot of globals in use in this file and it's getting quite hard to follow

Reply:
This is actually removed from this file entirely on the Oasis Direct branch, but I went ahead and made it more explicit here for the mean time